tests/src/core/testqgsvectorlayer.cpp

```diff
@@ -495,6 +496,39 @@ void TestQgsVectorLayer::testFieldAggregateExpression()
   QVERIFY( qgsDoubleNear( feature2.attribute( vfIndex ).toDouble(), 359065580.0, 1 ) );
 }
 
+void TestQgsVectorLayer::testAddFeatureExtentUpdated()
+{
+  QgsVectorLayer *layerLine = new QgsVectorLayer( QStringLiteral( "LineString?crs=EPSG:27700" ), QStringLiteral( "layer line" ), QStringLiteral( "memory" ) );
+  QVERIFY( layerLine->isValid() );
+  QCOMPARE( layerLine->featureCount(), static_cast<long>( 0 ) );
+  QCOMPARE( layerLine->extent(), QgsRectangle() );
+
+  const QSignalSpy spy_added( layerLine, &QgsVectorLayer::featureAdded );
+  const QSignalSpy spy_deleted( layerLine, &QgsVectorLayer::featureDeleted );
+
+  QgsFeature lineF1;
+  lineF1.setGeometry( QgsGeometry::fromWkt( QStringLiteral( "LineString (2 1, 1 1, 1 5, 7 12)" ) ) );
+
+  layerLine->startEditing();
+  layerLine->addFeature( lineF1 );
```

Comment:
I think to really test this fix we'd need to connect to featureAdded, and from THAT slot check the extent -- the test here doesn't confirm that a slot connected to featureAdded will have the correct extent available to it. (I suspect this test would pass under master without your fix)